Userland/rm.cpp

```diff
@@ -1,18 +1,62 @@
+#include <AK/AKString.h>
+#include <AK/Vector.h>
+#include <LibCore/CArgsParser.h>
+#include <LibCore/CDirIterator.h>
+#include <dirent.h>
 #include <stdio.h>
-#include <errno.h>
+#include <string.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
-int main(int argc, char** argv)
+int remove(bool recursive, const char* path)
 {
-    if (argc != 2) {
-        fprintf(stderr, "usage: rm <path>\n");
-        return 1;
-    }
-    int rc = unlink(argv[1]);
-    if (rc < 0) {
-        perror("unlink");
+    struct stat path_stat;
+    int s = stat(path, &path_stat);
+    if (s == -1)
         return 1;
+
+    if (S_ISDIR(path_stat.st_mode) && recursive) {
+        DIR* derp = opendir(path);
+        if (!derp) {
+            return 1;
+        }
+
+        while (auto* de = readdir(derp)) {
+            if (strcmp(de->d_name, ".") != 0 && strcmp(de->d_name, "..") != 0) {
+                char strbuf[256];
```

Comment:
This is a pretty small buffer. Use a `StringBuilder` instead and you don't have to worry about things fitting :)

```cpp
StringBuilder builder;
builder.append(path);
builder.append('/');
builder.append(de->d_name);
int s = remove(true, builder.to_string().characters());
...
```